### PR TITLE
added contour plotting method in sympy/plotting/plot.py

### LINKED ISSUE #12034: Add contour plot function
(Mentioned on [gitter](https://gitter.im/sympy/sympy?at=58740606cbcb2817704aa5dd), no responses yet.)

I saw 2d plots mentioned on [this wiki page](https://github.com/sympy/sympy/wiki/Plotting-capabilities#types-of-plots), but I understand that might be outdated/inaccurate.

Looking through what `plot3d()` does, I used `Plot(ContourSeries(…)).show()` as a workaround.
It seems like much of the infrastructure is in place to support a contour plotting function, would there be any difficulty doing so?

### PR DESCRIPTION
**Pull request by avishrivastava11**
Added contour plotting method in sympy/plotting/plot.py and modified test_plot.py for contour plotting

#12034
Fixes #12034
Took help and reviews from #12049

For documentation, reused the documentation of `plot3d`; however the examples were changed
Also, `plot3d` documents arguments taken on behalf of the `SurfaceOver2DRangeSeries` constructor, whereas `conplot` documents arguments using `ContourSeries`constructor.
The comment under class `ContourSeries` is updated as well now that there is contour plotting function that makes use of it.
In `test_plot.py` 3 tests have been added providing basis for the 3 examples present in the function's documentation. 

Note : 
I took the reviews of the PR made by @ArifAhmed1995 (https://github.com/sympy/sympy/pull/12049) and made this PR. I also took ideas from #12049 ( working code ).
Thus I have implemented the suggestions made there ( PR #12049 ) by @chrstphrchvz and @asmeurer 

Pls review this PR. Thank you.

### CODE REVIEW

**Comment by avishrivastava11** [comment]
@asmeurer @jksuom  can you review this PR ? and also, can you re-run the travis-build ? (I think there must've been some problem with my build, not the code, and that's why one test is having an issue)

**Comment by chrstphrchvz** [comment]
@avishrivastava11 Thanks for reviving this feature request, I'll try it out and give feedback within the next day or so.

**Comment by avishrivastava11** [comment]
@chrstphrchvz  sure, thanks.

@asmeurer  Can you take a look at this PR and suggest improvements?

**Comment by avishrivastava11** [comment]
@chrstphrchvz  Did you find the time to review this PR and give a feedback?

**Comment by avishrivastava11** [comment]
@jksuom @smichr @certik Can you review this PR ?

**Comment by chrstphrchvz** [comment]
@avishrivastava11 Seems to work as expected, thanks!

Examples from the test_plot.py in 506826370abf7aeb46287165998dbef37e5479e9:
```
plot_contour(sin(x)*sin(y), (x, -5, 5), (y, -5, 5))
```
image

```
plot_contour(x**2 + y**2, x**3 + y**3, (x, -5, 5), (y, -5, 5))
```
image

```
plot_contour((x**2 + y**2, (x, -5, 5), (y, -5, 5)), (x**3 + y**3, (x, -3, 3), (y, -3, 3)))
```
image

File: sympy/plotting/plot.py [hunk 1]
```diff
@@ -820,8 +820,7 @@ def get_meshes(self):
 class ContourSeries(BaseSeries):
     """Representation for a contour plot."""
     #The code is mostly repetition of SurfaceOver2DRange.
-    #XXX: Presently not used in any of those functions.
-    #XXX: Add contour plot and use this seties.
+    #XXX: Presently used in contour_plot function
```
Comment (certik) [inline sympy/plotting/plot.py]: Maybe you can remove the XXX, since there all the TODOs are done?

**Comment by certik** [comment]
@chrstphrchvz do you have any other suggestion to the PR?

File: sympy/plotting/plot.py [hunk 2]
```diff
@@ -820,7 +820,7 @@ def get_meshes(self):
 class ContourSeries(BaseSeries):
     """Representation for a contour plot."""
     #The code is mostly repetition of SurfaceOver2DRange.
-    #XXX: Presently used in contour_plot function
+    #Presently used in contour_plot function
```
Comment (Abdullahjavednesar) [inline sympy/plotting/plot.py]: Whitespace after #.

**Comment by Abdullahjavednesar** [comment]
@avishrivastava11 just those minor changes otherwise the changes looks good.

File: sympy/plotting/plot.py (hunk 2 above)
Comment (avishrivastava11) [reply sympy/plotting/plot.py]: @Abdullahjavednesar  Inculcated your review in the code. Thanks for the input.

**Comment by avishrivastava11** [comment]
@certik @Abdullahjavednesar  I have added the changes requested by you and @certik  to the code.
Waiting for completion of tests. If there are any more suggestions please let me know. Thanks.

File: sympy/plotting/plot.py (hunk 1 above)
Comment (avishrivastava11) [reply sympy/plotting/plot.py]: @certik  Added the changes. Thanks for the input.

**Review by certik — APPROVED** [review]
This looks good to me. @Abdullahjavednesar please go ahead and merge this once tests pass.

**Comment by chrstphrchvz** [comment]
I haven't thought of anything else for this PR.

However I do remember filing a [related issue](https://github.com/sympy/sympy/issues/12100) for wanting access to the `ContourSet` returned by [`contour()`](https://github.com/sympy/sympy/blob/03268800330a27da9e43ebba9eb58c0b00683261/sympy/plotting/plot.py#L913) to allow customizations like adding `clabel`s, in case anyone's interested in examining that again.

**Comment by chrstphrchvz** [comment]
Also, it probably would be good if the examples could be kept in the documentation, though I have no idea why Travis CI didn't like those being included. It might be helpful to squash the commits here, but that's up to the sympy team—I'm not familiar with this repo's commit/PR guidelines.

**Comment by avishrivastava11** [comment]
@chrstphrchvz 
>in case anyone's interested in examining that again.

I will look into it.

>Also, it probably would be good if the examples could be kept in the documentation, though I have no idea why Travis CI didn't like those being included

Yeah. I had initially kept those in the examples (in the documentation) but Travis consistently gave errors in that. So instead, I put those up in the tests.

I hope that's not much of a problem.

**Comment by avishrivastava11** [comment]
@Abdullahjavednesar @certik  One of the tests is failing due to some build issue ( there's no issue in the code as you can see from the previous test 6171d69 and only two whitespaces have been added in the next commit i.e. test no. 6208b99 ) Please can you re-run the Travis build?

**Comment by avishrivastava11** [comment]
@jksuom @smichr @certik @asmeurer @Abdullahjavednesar , can you re-run the travis-build ? (I think there must've been some problem with my build, not the code, and that's why two tests are having an issue)

**Comment by avishrivastava11** [comment]
@certik 
>This looks good to me. @Abdullahjavednesar please go ahead and merge this once tests pass.

@Abdullahjavednesar merge this now?

**Comment by Abdullahjavednesar** [comment]
Thanks for the good work.

**Comment by avishrivastava11** [comment]
Thanks @certik @Abdullahjavednesar @chrstphrchvz 
This is my first merged PR.
Great experience.
:)

**Pull request merged**